lib/base_16.rb

```diff
@@ -0,0 +1,16 @@
+class Base16
+
+  # The IRS has requested data be encoded this way. Loosely emulate the Base64 class.
+
+  def self.encode16(str)
+    str.bytes.map { |char| char.to_s(16).upcase.rjust(2, "0") }.join
+  end
+
+  def self.decode16(str)
+    output = ''
+    str.chars.each_slice(2) do |chars|
+      output << chars.join.to_i(16).chr
+    end
+    output
+  end
```

Comment:
two thoughts:
1. this gem exists: https://github.com/sspreitzer/base16
2. we're going to be doing this on 50MB files-ish at a time, I am very curious to see what optimizations we can make, if any so we're not constantly re-allocating strings, or if there are versions that can work on a stream or on a file, etc

Reply:
Ok so initial results are in! this is faster than the gem so let's leave it, but I think it's ripe for further optimization

```
ruby base16.rb
Warming up --------------------------------------
          encode_gem     4.000  i/100ms
           encode_pr    17.000  i/100ms
Calculating -------------------------------------
          encode_gem     31.651  (±15.8%) i/s -    156.000  in   5.062737s
           encode_pr    253.741  (± 2.8%) i/s -      1.275k in   5.029016s

Comparison:
           encode_pr:      253.7 i/s
          encode_gem:       31.7 i/s - 8.02x  (± 0.00) slower

Warming up --------------------------------------
          decode_gem     5.000  i/100ms
           decode_pr    14.000  i/100ms
Calculating -------------------------------------
          decode_gem     47.739  (±14.7%) i/s -    230.000  in   4.999974s
           decode_pr    142.206  (± 3.5%) i/s -    714.000  in   5.028501s

Comparison:
           decode_pr:      142.2 i/s
          decode_gem:       47.7 i/s - 2.98x  (± 0.00) slower

```

Reply:
see https://github.com/18F/identity-idp/pull/7274